packages/instantsearch-react/src/utils.js

```diff
@@ -12,3 +12,20 @@ export function isSpecialClick(event) {
 export function capitalize(key) {
   return key.length === 0 ? '' : `${key[0].toUpperCase()}${key.slice(1)}`;
 }
+
+export function assertFacetDefined(searchParameters, searchResults, facet) {
+  const wasRequested =
+    searchParameters.isConjunctiveFacet(facet) ||
+    searchParameters.isDisjunctiveFacet(facet);
+  const wasReceived =
+    Boolean(searchResults.getFacetByName(facet));
+  if (searchResults.nbHits > 0 && wasRequested && !wasReceived) {
```

Comment:
When `nbHits == 0`, the API won't send any facets, so `wasReceived` is always false.